Doc/library/stdtypes.rst

```diff
@@ -2841,6 +2841,13 @@ objects.
          optional *sep* and *bytes_per_sep* parameters to insert separators
          between bytes in the hex output.
 
+   .. method:: resize(size)
+
+      Resize the :class:`bytearray` to contain *size* bytes.
+      If :class:`bytearray` needs to grow, all new bytes will be set to null bytes.
```

Comment:
Also describe the "obvious-to-us" behavior of what happens when it shrinks: the data at the end is truncated, the remaining data should be equivalent to a `[:size]` slice.

... and consider if the Python version should also support negative size to mean the same thing it would in a slice notation.
`if (size < 0): size = max(0, len(self) + size)` ?

Reply:
:+1: for the truncation.

I don't like negative, because the function operates in absolute buffer size, and requesting a negative buffer sign sounds like a bug I'd write and I'd prefer an exception / exit rather than debug the symptom "my buffer shrank".

Reply:
my original equivalent to is wrong, which make it look like a delta, current doc one I'm validating:
```python
if len(self) > size:
    del self[size:]
else:
    self += b'\0' * (size - len(self))
```